src/attr.rs

```diff
@@ -24,12 +27,90 @@ pub fn valid<I: Iterator<Item = char>>(chars: I) -> (usize, bool) {
     (p.pos, has_attr)
 }
 
+/// Stores an attribute value that supports backslash escapes of ASCII punctuation upon displaying,
+/// without allocating.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct AttributeValue<'s> {
```

Comment:
since the module is called `attr` we could perhaps call this `attr::Value` instead of `attr::AttributeValue`? same with `attr::ValueParts` in that case.

Reply:
Reasonable, indeed! Although then perhaps we should not expose the structs of the `attr` module directly to the library root? Having `Value` and `ValueParts` in the top-level of the docs could be a bit confusing (same applies for `use` statements.)

Reply:
Oh, yes, true, I forgot to consider the public API. I don't think we want to expose the `attr` module, as it seems like an implementation detail. Simply exposing `Attributes`, `AttributeValue` and `AttributeValueParts` is probably best. We could either keep the full name `AttributeValue` or do `pub use attr::Value as AttributeValue`. I think the the first option might be best, because then it is clear that only one type exists.

Reply:
Done!